src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -63,21 +64,25 @@ class WorkspaceService(protected val argUserToken: WithAccessToken,
 
   implicit val userToken: WithAccessToken = argUserToken
 
+  val priceList = Map("STANDARD" -> 0.02,
+                      "NEARLINE" -> 0.01,
+                      "COLDLINE" -> 0.004,
+                      "ARCHIVE" -> 0.0012,
+                      "REGIONAL" -> 0.02,
+                      "MULTI-REGIONAL" -> 0.02
+  ) // TODO: dra?
+
   def getStorageCostEstimate(workspaceNamespace: String,
                              workspaceName: String,
                              userProject: Option[GoogleProjectId]
   ): Future[RequestComplete[WorkspaceStorageCostEstimate]] = for {
-    bucketUsage <- rawlsDAO.getBucketUsage(workspaceNamespace, workspaceName)
-    priceList <- googleServicesDAO.fetchPriceList
-    bucketOptions <- rawlsDAO.getBucketOptions(workspaceNamespace, workspaceName, userProject)
+    bucketUsage <- rawlsDAO.getBucketUsageV2(workspaceNamespace, workspaceName)
   } yield {
-    val rate = priceList.prices.cpBigstoreStorage.getOrElse(
-      bucketOptions.location.toLowerCase,
-      priceList.prices.cpBigstoreStorage("us")
-    )
     // Convert bytes to GB since rate is based on GB.
-    val estimate: BigDecimal = BigDecimal(bucketUsage.usageInBytes) / (1024 * 1024 * 1024) * rate
-    RequestComplete(WorkspaceStorageCostEstimate(f"$$$estimate%.2f", bucketUsage.lastUpdated))
+    val estimate: BigDecimal = bucketUsage.metrics
+      .map(metric => BigDecimal(metric.valueInBytes) / (1024 * 1024 * 1024) * priceList(metric.storageClass))
+      .sum
+    RequestComplete(WorkspaceStorageCostEstimate(f"$$$estimate%.2f", Some(DateTime.now())))
```

Comment:
see my previous comment on reference.conf; this probably needs to re-include the bucket location for an accurate price.